docs/reference/io.rst

```diff
@@ -1,48 +1,56 @@
 
+The primary focus of ``sunpy.io`` is to provide a common interface for reading and writing.
 ``sunpy.io`` contains readers for files that are commonly used in solar physics.
 
-Unified File Readers
-====================
+These include:
 
-.. automodapi:: sunpy.io
+- Public API
+  - GENX
+  - ANA
+  - NOAA SWPC Solar Region Summary (SRS)
+  - ASDF
 
-.. automodapi:: sunpy.io.ana
+- Internal/Private API
+  - FITS
+  - JPEG2000
+  - CDF
+
+However, only the special file readers are documented here as they public API.
+
+The other readers are intended for use by `sunpy.map` and `sunpy.timeseries` and are not intended to be used directly by users.
 
 Special File Readers
 ====================
 
 .. automodapi:: sunpy.io.special.genx
 
-``sunpy`` has a custom reader for NOAA SWPC Solar Region Summary (SRS) files:
+.. automodapi:: sunpy.io.ana
 
 .. automodapi:: sunpy.io.special.srs
 
 asdf (Advanced Scientific Data Format)
 --------------------------------------
 
-`ASDF <https://asdf-standard.readthedocs.io/en/latest/>`__ is a modern file format
-designed to meet the needs of the astronomy community. It has deep integration
-with Python, SunPy, and Astropy, as well as implementations in other languages.
-It can be used to store known Python objects in a portable, well defined file
-format. It is primarily useful for storing complex Astropy and SunPy objects in
-a way that can be loaded back into the same form as they were saved.
-It is designed to be an archive file format, with human readable metadata and a
-simple on-disk layout.
+`ASDF <https://asdf-standard.readthedocs.io/en/latest/>`__ is a modern file format designed to meet the needs of the astronomy community [citation needed].
+It has deep integration with Python, sunpy, and Astropy, as well as implementations in other languages.
+It can be used to store known Python objects in a portable, well defined file format.
+It is primarily useful for storing complex Astropy and sunpy objects in a way that can be loaded back into the same form as they were saved.
+It is designed to be an archive file format, with human readable metadata and a simple on-disk layout.
 
-sunpy currently implements support for saving `Map <sunpy.map.GenericMap>` and
-`coordinate frame <sunpy.coordinates.frames>` objects into asdf files. As asdf
-tightly integrates into Python, saving a map to an asdf file will save the
-metadata, data and mask. In comparison, the mask is not currently saved
-to FITS. The following code shows to to save and load a sunpy Map to an asdf
-file
+sunpy currently implements support for saving `Map <sunpy.map.GenericMap>` and `coordinate frame <sunpy.coordinates.frames>` objects into asdf files.
+As asdf tightly integrates into Python, saving a map to an asdf file will save the metadata, data and mask.
+In comparison, the mask is not currently saved to FITS.
+The following code shows to to save and load a sunpy Map to an asdf file:
 
-.. doctest-requires:: asdf
+.. code-block:: python
```

Comment:
This reads more like a how-to guide. I don't think it belongs in our reference material.

Reply:
I will make it one.